src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -3,19 +3,103 @@
 import org.junit.Before;
 import org.junit.Test;
 
+import java.math.BigDecimal;
+import java.util.Currency;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
 
-public class ArrayListProductDaoTest
-{
+public class ArrayListProductDaoTest {
     private ProductDao productDao;
+    private Currency usd;
 
     @Before
     public void setup() {
         productDao = new ArrayListProductDao();
+        usd = Currency.getInstance("USD");
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowExceptionWhenSaveNullProduct() {
+        productDao.save(null);
+    }
+
+    @Test
+    public void shouldSaveNewProduct() {
+        Product product = new Product("test-product1", "Test Product 1", new BigDecimal(100), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg");
```

Comment:
"test-product1", 100, etc are magic variables, please make them constant. Fix this issue for the whole class.